File: fizzbuzz.js
//FizzBuzz algorithm problem:

//Solution 1:
#Rule: For numbers divisible by 3, print out "Fizz".
      #For numbers divisible by 5, print out "Buzz".
      #For numbers divisible by both 3 and 5, print out "FizzBuzz" in the console.
      #Otherwise, just print out the number.
      
var fizzBuzz = function(num) {
    for(var i = 1; i <= num; i++){
        if(i % 3 === 0 && i % 5 === 0){
            console.log('FizzBuzz');
        } else if (i % 5 === 0){
            console.log('Buzz');
        } else if (i % 3 === 0){
            console.log('Fizz');
        } else {
            console.log(i);
        }
    }
}

fizzBuzz(100);

//Solution 2:
//Require NodeJS & Readline-Sync

var readlineSync = require('readline-sync');

var input = readlineSync.question('How many digit of FizzBuzz shall I produce: ');

for (var iteration = 1; iteration <= Number(input); iteration++) {
  if (iteration % 3 === 0 && iteration % 5 === 0) {
    console.log('FizzBuzz');
  } else if (iteration % 3 === 0) {
    console.log('Fizz');
  } else if (iteration % 5 === 0) {
    console.log('Buzz');
  } else {
    console.log(iteration);
  }
}

File: fizzbuzz.rb
#Foolbar coding challenge
#Rule 1: For all numbers that are divisible by three, the player should say "Fizz."
#Rule 2: For any number divisible by five, the player should say "Buzz."
#Rule 3: Numbers divisible by both become "FizzBuzz".


#prompt user for a number of pattern they want to see
puts "How many digits of pattern you want to see?"
digits =gets.to_i

n = 0

while n < digits
  n = n+1
#fizzbuzz logic:    
    if n % 3 == 0 && n % 5 == 0
      puts "FizzBuzz"
    elsif n % 3 == 0
      puts "Fizz"
    elsif n % 5 == 0 
      puts "Buzz"
    else
      puts "#{n}"
    end
end

#Solution 2:
def fizzbuzz(num)
  i = 0
  while i < num
  i += 1
  
    if i % 3 == 0 && i % 5 == 0
      puts "FizzBuzz"
    elsif i % 5 == 0
      puts "Buzz"
    elsif i % 3 == 0
      puts "Fizz"
    else
      puts i
    end
  end
end
